lib/repository.js

```diff
@@ -132,7 +132,22 @@ Repository.prototype.save = function (aggregate, commitId, callback) {
               LOG.warn('IGNORING SNAPSHOT STORE. VERSION MISMATCH MIGHT LEAD TO SNAPSHOT FAILURE. for stream %s version %s - start stream version: %s - start aggregate version: %s', aggregate.id, aggregate.getVersion(), startStreamVersion, startAggregateVersion);
             } else {
               LOG.debug('Persisting snapshot for stream %s version %s - start stream version: %s - start aggregate version: %s', aggregate.id, aggregate.getVersion(), startStreamVersion, startAggregateVersion);
-              self._partition.storeSnapshot(aggregate.id, aggregate._getSnapshot(), aggregate.getVersion());
+              const SNAPSHOT_RETRY_TIMEOUT = 5000;
+              self._partition.storeSnapshot(aggregate.id, aggregate._getSnapshot(), aggregate.getVersion()).catch((e) => {
+                LOG.error('Error storing snapshot. findBySnapshot and retry once... with error: %j', e);
+                setTimeout(() => {
+                  const onError = (retryError) => {
+                    LOG.error('Error storing snapshot for stream %s with error: %j', aggregate.id, retryError);
+                  };
+                  try {
+                    self.findBySnapshot(id, false).then((newAggregate) => {
+                      self._partition.storeSnapshot(newAggregate.id, newAggregate._getSnapshot(), newAggregate.getVersion()).catch(onError);
+                    });
+                  } catch (e) {
+                    onError();
+                  }
```

Comment:
I think it might work. But why not just just the catch of the findBySnapShot instead of try-catch?

```suggestion
                  self.findBySnapshot(id, false)
                    .then((newAggregate) => {
                      self._partition.storeSnapshot(newAggregate.id, newAggregate._getSnapshot(), newAggregate.getVersion())
                        .catch(onError);
                    })
                    .catch(onError);
```